<launch>

  <group ns="perception"> 
    <arg name="camera_config_file" value="$(find percept)/config/static_cameras.yaml"/>
    <rosparam file="$(arg camera_config_file)" command="load"/>

    <arg name="pipeline_config_file" value="$(find percept)/config/pipeline.yaml"/>
    <rosparam file="$(arg pipeline_config_file)" command="load"/>

    <node name="perception_node" pkg="percept" type="real_time_pipeline.py" args="--static" output="screen"/>    
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find percept)/rviz/perception.rviz"/>
  </group>


  <group ns="cameras">

    <group ns="camera_1">
      <arg name="serial_no"           default="032622070733"/>
      <arg name="camera"              default="camera_1"/>
      <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
        <arg name="serial_no"         value="$(arg serial_no)"/>
        <arg name="enable_depth"      value="true"/>
        <arg name="depth_width"       value="640"/>
        <arg name="depth_height"      value="360"/>
        <arg name="depth_fps"         value="30"/>
        <arg name="enable_pointcloud" value="true"/>
        <arg name="pointcloud_texture_stream" default="RS2_STREAM_ANY"/> 
        <arg name="tf_prefix"         value="$(arg camera)"/> 
        <arg name="enable_sync"       value="true"/>
        <arg name="enable_color"      value="false"/>    
        <arg name="color_fps"         value="0"/>
      </include>
    </group>

  </group>
  
</launch>